src/main/java/com/hillel/vynnyk/homeworks/homework13/NewArrayInitializer.java

```diff
@@ -0,0 +1,37 @@
+package com.hillel.vynnyk.homeworks.homework13;
+
+import java.util.concurrent.RecursiveAction;
+
+public class NewArrayInitializer extends RecursiveAction {
+    private final double[] array;
+    private final int start;
+    private final int end;
+
+
+
+    public NewArrayInitializer(double[] array, int start, int end) {
+        this.array = array;
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    protected void compute() {
+        if (end - start <= 6) {
+            for (int i = 0; i < array.length; i++) {
+                array[i] = array[i] * Math.sin(0.2 + i / 5.0) * Math.cos(0.2 + i / 5.0) * Math.cos(0.4 + i / 2.0);
+            }
+        } else {
+            NewArrayInitializer firstTask = new NewArrayInitializer(array, 0, array.length / 2);
```

Comment:
лише зараз звернув увагу, це буде працювати за умови однієї ітерації. 
за другої вже будуть проблеми. 
```suggestion
            int mid = (start + end) / 2; 
            NewArrayInitializer firstTask = new NewArrayInitializer(array, start, mid);
            NewArrayInitializer secondTask = new NewArrayInitializer(array, mid, end);
```